platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -136,7 +139,41 @@ private static boolean solrCoreExists(SolrClient client, String coreName)
       ConfigurationStore.getInstance().setDataDirectoryPath(solrDataDir);
     }
     LOGGER.debug("Solr({}): Creating an HTTP Solr client using url [{}]", core, coreUrl);
-    return new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+
+    SolrClientAdapter adaptor =
+        new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+    try {
+      if (!adaptor.isAvailable(30, 5, TimeUnit.SECONDS)) {
+        LOGGER.warn("Solr Client {} is not available after 30 seconds", core);
+      }
+    } catch (InterruptedException e) {
+      LOGGER.error("Unable to connect to solr client {}: {} ", core, e.getStackTrace());
```

Comment:
❗️ Improper way to handle interruptions. They need to be propagated up by re-interrupting the current thread.